spock-core/src/main/java/org/spockframework/runtime/model/MethodInfo.java

```diff
@@ -122,12 +124,23 @@
   public Object invoke(Object target, Object... arguments) throws Throwable {
     for (int i = 0, argCount = arguments.length; i < argCount; i++) {
       if (arguments[i] == MISSING_ARGUMENT) {
-        StringJoiner missingArguments = new StringJoiner("', '", "No argument was provided for parameters: '", "'");
-        List<String> parameterNames = getFeature().getParameterNames();
-        missingArguments.add(parameterNames.get(i));
-        for (int j = i + 1; j < argCount; j++) {
-          if (arguments[j] == MISSING_ARGUMENT) {
-            missingArguments.add(parameterNames.get(j));
+        StringJoiner missingArguments;
+        if (getKind() == FEATURE) {
+          missingArguments = new StringJoiner("', '", "No argument was provided for parameters: '", "'");
```

Comment:
both are initialized with the same value, so you could move it up to the declaration

Reply:
No, they are not.
The ones with names get quotes around, the ones with indexes not.

Reply:
Well, with `parameters[i].getName()` now, I made both cases with quotes, so now they are the same again. :-)